python/tvm/script/parser.py

```diff
@@ -1110,6 +1145,32 @@ def transform_TypeConstant(self, node):
         """
         return node.value
 
+    def transform_TypeTuple(self, node):
+        """Tuple value visitor for types.
+
+        Mostly used in `transform_TypeCall` and `transform_TypeApply`.
+        """
+        return [self.transform(value) for value in node.values]
+
+    def handle_match_buffer_type(self, node, buffer_name):
+        """special function to handle syntax sugar for match buffer.
+
+        This method is for syntax sugar of T.match_buffer()/[]
+        """
+        func = self.transform(node.func_name)
+
+        if isinstance(func, SpecialStmt):
```

Comment:
since this method is now specifically for match buffer, you can assert this and remove the error reporting below